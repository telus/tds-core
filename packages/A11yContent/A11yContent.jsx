import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import { safeRest } from '@tds/util-helpers'
import { componentWithName, or } from '@tds/util-prop-types'

const StyledA11yContent = styled.span({
  position: 'absolute',
  height: '1px',
  width: '1px',
  overflow: 'hidden',
  clip: 'rect(1px, 1px, 1px, 1px)',
})

/**
 * @version ./package.json
 */

const A11yContent = ({ children, ...rest }) => {
  return <StyledA11yContent {...safeRest(rest)}>{children}</StyledA11yContent>
}

A11yContent.propTypes = {
  /**
   * Accessible content. Can be either a string or a `Heading` Component
   */
  children: or([PropTypes.string, componentWithName('Heading')]).isRequired,
}
A11yContent.defaultProps = {}

export default A11yContent
