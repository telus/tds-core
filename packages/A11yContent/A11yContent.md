Allows content to be hidden from sight, but still visible to screen readers. This can be used to give context to users using screen readers without revealing unnecessary extra text to sighted users.

```jsx
    <Text>The screen reader says:</Text><A11yContent>This is accessible content</A11yContent>
```

If `A11yContent` has to go between two separate text components, considering wrapping the whole text in a `<span>` to persist leading and trailing space.

```jsx
<ButtonLink href="#">
  <span>
    Shop <A11yContent>iPhone</A11yContent> Now
  </span>
</ButtonLink>
```
