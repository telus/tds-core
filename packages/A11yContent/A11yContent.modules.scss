.a11yContent {
  position: absolute;
  height: 1px;
  width: 1px;
  overflow: hidden;
  clip: rect(1px, 1px, 1px, 1px);
}
