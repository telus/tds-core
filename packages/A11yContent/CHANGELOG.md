# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 2.3.3 (2022-02-16)


### Bug Fixes

* gha release job npm publish token ([ce7aacd](https://github.com/telusdigital/tds/commit/ce7aacdd77e54c4f8e48eb31d0c05dc804530324))





## 2.3.2 (2022-02-15)


### Bug Fixes

* gha release job git config ([276ebb6](https://github.com/telusdigital/tds/commit/276ebb6968a0b56c9b87c178f6895a60ae108e71))





## 2.3.1 (2022-02-15)


### Bug Fixes

* gha release job git config ([1ab603d](https://github.com/telusdigital/tds/commit/1ab603d68c36219b0711fc353bc2515b64712ca9))





# [2.3.0](https://github.com/telusdigital/tds/compare/@tds/core-a11y-content@2.2.4...@tds/core-a11y-content@2.3.0) (2021-09-21)


### Features

* **core-a11y-content:** add react 17 to peer deps ([6ddb23f](https://github.com/telusdigital/tds/commit/6ddb23fed7cb003761d70844fac3f40865731fa0))





## [2.2.4](https://github.com/telusdigital/tds/compare/@tds/core-a11y-content@2.2.3...@tds/core-a11y-content@2.2.4) (2021-01-15)

**Note:** Version bump only for package @tds/core-a11y-content





## [2.2.3](https://github.com/telusdigital/tds/compare/@tds/core-a11y-content@2.2.2...@tds/core-a11y-content@2.2.3) (2020-09-11)

**Note:** Version bump only for package @tds/core-a11y-content





## [2.2.2](https://github.com/telusdigital/tds/compare/@tds/core-a11y-content@2.2.1...@tds/core-a11y-content@2.2.2) (2020-09-08)


### Bug Fixes

* **core-a11y-content:** add support for styled-components v5 ([d665a98](https://github.com/telusdigital/tds/commit/d665a98e85c91eb7b111e8230c767c0d7fbe04f5))





## [2.2.1](https://github.com/telusdigital/tds/compare/@tds/core-a11y-content@2.2.0...@tds/core-a11y-content@2.2.1) (2020-09-04)

**Note:** Version bump only for package @tds/core-a11y-content





# [2.2.0](https://github.com/telusdigital/tds/compare/@tds/core-a11y-content@2.1.5...@tds/core-a11y-content@2.2.0) (2020-08-07)


### Features

* **core-a11y-content:** add tds-heading as valid child prop-type ([fdd194b](https://github.com/telusdigital/tds/commit/fdd194b6ac24f4d9111d198af55a5db2edb450f4))





## [2.1.5](https://github.com/telusdigital/tds/compare/@tds/core-a11y-content@2.1.4...@tds/core-a11y-content@2.1.5) (2020-02-03)

**Note:** Version bump only for package @tds/core-a11y-content





## [2.1.4](https://github.com/telusdigital/tds/compare/@tds/core-a11y-content@2.1.3...@tds/core-a11y-content@2.1.4) (2020-01-27)

**Note:** Version bump only for package @tds/core-a11y-content





## [2.1.3](https://github.com/telusdigital/tds/compare/@tds/core-a11y-content@2.1.2...@tds/core-a11y-content@2.1.3) (2020-01-23)

**Note:** Version bump only for package @tds/core-a11y-content





## [2.1.2](https://github.com/telusdigital/tds/compare/@tds/core-a11y-content@2.1.1...@tds/core-a11y-content@2.1.2) (2020-01-22)

**Note:** Version bump only for package @tds/core-a11y-content





## [2.1.1](https://github.com/telusdigital/tds/compare/@tds/core-a11y-content@2.1.0...@tds/core-a11y-content@2.1.1) (2019-09-06)

**Note:** Version bump only for package @tds/core-a11y-content





# [2.1.0](https://github.com/telusdigital/tds/compare/@tds/core-a11y-content@2.0.3...@tds/core-a11y-content@2.1.0) (2019-06-06)


### Bug Fixes

* **core-a11y-content:** change peerDependency ([bf38195](https://github.com/telusdigital/tds/commit/bf38195))


### Features

* **core-a11y-content:** enable rest props ([eae8b41](https://github.com/telusdigital/tds/commit/eae8b41))





## [2.0.3](https://github.com/telusdigital/tds/compare/@tds/core-a11y-content@2.0.2...@tds/core-a11y-content@2.0.3) (2019-05-01)

**Note:** Version bump only for package @tds/core-a11y-content





## [2.0.2](https://github.com/telusdigital/tds/compare/@tds/core-a11y-content@2.0.1...@tds/core-a11y-content@2.0.2) (2019-04-23)

**Note:** Version bump only for package @tds/core-a11y-content





## [2.0.1](https://github.com/telusdigital/tds/compare/@tds/core-a11y-content@2.0.0...@tds/core-a11y-content@2.0.1) (2019-04-22)

**Note:** Version bump only for package @tds/core-a11y-content





# [2.0.0](https://github.com/telusdigital/tds/compare/@tds/core-a11y-content@1.0.1...@tds/core-a11y-content@2.0.0) (2019-03-28)


### Features

* **core-a11y-content:** upgrade to styled-components ([2596e66](https://github.com/telusdigital/tds/commit/2596e66))


### BREAKING CHANGES

* **core-a11y-content:** upgrade to styled-components





## [1.0.1](https://github.com/telusdigital/tds/compare/@tds/core-a11y-content@1.0.0...@tds/core-a11y-content@1.0.1) (2019-02-06)

**Note:** Version bump only for package @tds/core-a11y-content





<a name="1.0.0"></a>
# 1.0.0 (2018-07-17)


### Bug Fixes

* **core-a11y-content:** change children PropType to string ([4efdcd0](https://github.com/telusdigital/tds/commit/4efdcd0))


### Features

* **core-a11y-content:** add a11y-content component base ([d177097](https://github.com/telusdigital/tds/commit/d177097))
* **core-button-link:** add A11yContent props ([7867a41](https://github.com/telusdigital/tds/commit/7867a41))


### BREAKING CHANGES

* **core-a11y-content:** Publishes this as 1.0.0!
