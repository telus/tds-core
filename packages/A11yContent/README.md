# TDS Core: A11yContent

Hide text visually, but retain visibility to screen readers.
