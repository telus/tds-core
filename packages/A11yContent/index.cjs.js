require('./dist/index.css')
const A11yContent = require('./dist/index.cjs')

module.exports = A11yContent
