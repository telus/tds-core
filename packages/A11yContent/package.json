{
  "name": "@tds/core-a11y-content",
  "version": "2.1.5",
  "description": "Hide content that should only be seen by screen readers",
  "main": "index.cjs.js",
  "module": "index.es.js",
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/telusdigital/tds.git"
  },
  "publishConfig": {
    "access": "public"
  },
  "author": "TELUS digital",
  "engines": {
    "node": ">=8"
  },
  "bugs": {
    "url": "https://github.com/telusdigital/tds/issues"
  },
  "homepage": "http://tds.telus.com",
  "peerDependencies": {
    "react": "^16.8.2",
    "react-dom": "^16.8.2",
    "styled-components": "^4.1.3"
  },
  "dependencies": {
    "@tds/util-helpers": "^1.4.2",
    "@tds/util-prop-types": "^1.4.0",
    "prop-types": "^15.5.10"
  }
}
