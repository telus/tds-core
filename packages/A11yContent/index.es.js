import './dist/index.css'
import A11yContent from './dist/index.es'

export default A11yContent
