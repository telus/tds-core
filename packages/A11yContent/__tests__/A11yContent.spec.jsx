import React from 'react'
import { shallow } from 'enzyme'

import A11yContent from '../A11yContent'

describe('A11yContent', () => {
  const defaultProps = {}
  const doShallow = (props = {}) => {
    return shallow(
      <A11yContent {...defaultProps} {...props}>
        Some text
      </A11yContent>
    )
  }

  it('renders', () => {
    const a11yContent = doShallow()

    expect(a11yContent).toMatchSnapshot()
  })

  it('does not allow custom CSS', () => {
    const a11yContent = doShallow({ className: 'my-custom-class', style: { color: 'hotpink' } })

    expect(a11yContent).not.toHaveProp('className', 'my-custom-class')
    expect(a11yContent).not.toHaveProp('style')
  })
})
