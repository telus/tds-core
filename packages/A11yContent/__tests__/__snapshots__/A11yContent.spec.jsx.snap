// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A11yContent renders 1`] = `
<span
  className="a11yContent"
>
  Some text
</span>
`;
